src/charm.py

```diff
@@ -121,6 +125,25 @@ def _is_legacy_v1_relation(self, relation):
 
         return True
 
+    def _configure_ingress(self, event: HookEvent) -> None:
+        """Make sure the traefik route relation data is up to date."""
+        if not self.unit.is_leader():
+            return
```

Comment:
Do you mean this is "ingress to leader"?
Is there any cleanup we would need to do on `leader-settings-changed`?

Reply:
We do have `_on_leader_elected` for updating the relations: https://github.com/canonical/tempo-k8s-operator/blob/4f97ae8185d46062b48cbf9e5e101dce7deffff1/src/charm.py#L178